community/kernel/src/main/java/org/neo4j/kernel/api/KernelTransaction.java

```diff
@@ -108,9 +110,20 @@
     /**
      * Closes this transaction, committing its changes iff {@link #success()} has been called and
      * {@link #failure()} has NOT been called. Otherwise its changes will be rolled back.
+     *
+     * @return id of the committed transaction or {@link #NOT_COMMITTED} if transaction was rolled back or read-only.
```

Comment:
Changes will not be committed if `markForTermination()` has been called, even if `failure()` has not been called explicitly.